Automattic/wp-calypso
# Modularise NPS survey state

This PR is one of many working on modularising state in Calypso. This one handles NPS survey state.

For more details on state modularisation, see the [modularised state documentation](https://github.com/Automattic/wp-calypso/blob/master/docs/modularized-state.md) and p4TIVU-9lM-p2.

Note: I added all the reviewers that GitHub suggested. Please feel free to ignore the review request or pull in someone else if you're not comfortable reviewing this PR. Thank you!

Fixes #42463.

#### Changes proposed in this Pull Request

* Modularise NPS survey state
* Move `ui/nps-survey-notice` to `nps-survey/notice`.

#### Testing instructions

NPS survey state is always loaded on boot, since the NPS survey is used on the main logged-in layout. Therefore, while this change might have a positive impact in bundle sizes (by reducing the critical path bundle size), it's difficult to see the state going from unloaded to loaded, as it happens without user input on boot.

To verify that this change works correctly, please smoke test NPS survey functionality and ensure that it continues to work as before. I'm completely unfamiliar with the feature and what triggers it, so I'm afraid I can't provide further details 😕
